cmd/main.go

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/m3db/prometheus_remote_client_golang/promremote"
+)
+
+type tagList []promremote.Tag
+type dp promremote.Datapoint
+
+func main() {
```

Comment:
usability/invocation nit, but i'm not personally a fan of flag-only tools - there's no real correct answer, but the rationale behind my opinion is that a tool basically takes the form

    tool [<customization_flags>] [<command> [arg1, ...]]

so in this case, i would have `write` be the command, `now` be the default value of `-ts`, and leave tags as-is, so the final form would be:

    $ promremote write -t foo:bar -t baz:bat 1234.5
    $ promremote write -t foo:bar -ts 1234567890 1234.5

etc. it also might be a little weird to have comma-delimited `-d` pairs but `:`-delimited `-t` pairs (another reason i'd probably use positional args rather than flags for the datapoint).

also, what are your thoughts on using this in bulk (e.g. to emit N `val,ts` metrics with the same tags)? something like

    $ promremote write -t foo:bar -t baz:bat val1:ts1 val2:ts2

or similar?

if you're happy with how it works, that's fine too.

Reply:
I think as a v1 this is fine. I'm gonna work with one of the guys in OSS who wants this and see what is useful for him once he tries this out.

Reply:
Yeah agreed, let's revisit this in a follow up change.